assets/src/dashboard/app/views/myStories/content/storyGridItem/index.js

```diff
@@ -71,72 +67,124 @@ const StoryGridItem = ({
       }
     : {};
 
+  const isLocked = useMemo(() => {
+    return enablePostLocking && story?.locked && userId !== story?.lockUser.id;
+  }, [enablePostLocking, story, userId]);
+
   const generatedMenuItems = useMemo(
     () =>
       generateStoryMenu({
         menuItemActions: storyMenu.menuItemActions,
         menuItems: storyMenu.menuItems,
         story,
+        isLocked,
       }),
-    [storyMenu, story]
+    [storyMenu, story, isLocked]
   );
 
   const storyDate = getRelativeDisplayDate(
     story?.status === STORY_STATUS.DRAFT
       ? story?.modified_gmt
       : story?.created_gmt
   );
+
+  const formattedTitle = titleFormatted(story.title);
+
+  const memoizedStoryMenu = useMemo(
+    () => (
+      <StoryMenu
+        menuLabel={
```

Comment:
now when you key through stories grid with arrows you're just hitting theses menu buttons so updating them to give actual context.